app/components/enterprise-banner.js

```diff
@@ -0,0 +1,122 @@
+      this.get('ajax').get(url).then(response => {
+        const exp = new Date(Date.parse(response.expiration_time));
+        this.set('licenseId', response.license_id);
+        this.set('expirationTime', exp);
+        this.set('daysUntilExpiry', Math.ceil(
+          (exp.getTime() - new Date().getTime()) / (1000 * 60 * 60 * 24)
+        ));
+        this.set('licenseType', response.license_type);
+        this.set('seats', response.seats);
+        this.set('activeUsers', response.active_users);
+        this.set('isTrial', response.license_type === 'trial');
+        this.set('isPaid', response.license_type !== 'trial');
+        if (!this.get('expiring')) {
+          this.get('storage').removeItem(this.get('key'));
+        }
+      });
+    });
+  },
+
+  @computed('seats', 'activeUsers')
+  exceedingSeats(seats, activeUsers) {
+    return (activeUsers > seats);
+  },
+
+  @computed('seats', 'activeUsers')
+  almostExceedingSeats(seats, activeUsers) {
+    return (seats - activeUsers <= 5);
+  },
+
+  @computed('expirationTime')
+  isExpired(expirationTime) {
+    return new Date() > expirationTime;
+  },
+
+  @computed('expirationTime')
+  expirationTimeFromNow(expirationTime) {
+    return new Ember.String.htmlSafe(timeAgoInWords(expirationTime) || '-');
+  },
+
+  @computed('daysUntilExpiry')
+  expiring(days) {
+    if (!days) return false;
+    return days <= 60;
+  },
+
+  @computed('daysUntilExpiry')
+  expiringHalfway(days) {
+    if (!days) return false;
+    return days <= 30;
+  },
+
+  @computed('daysUntilExpiry')
+  expiringSoon(days) {
+    if (!days) return false;
+    return days <= 10;
+  },
+
+  @computed('expiring', 'expiringHalfway', 'expiringSoon')
+  checkLicenseBanner(expiring, halfway, soon) {
+    let lastSeen = this.get('storage').getItem(this.get('key'));
+    if (
+      // User has never closed banner, and license expires in 60 days or less
+      (!lastSeen && expiring) ||
+      // User has either never closed the banner, or closed it before 30 days,
+      // and license expires in 30 days or less
+      ((!lastSeen || lastSeen > 30) && halfway) ||
+      // License expires in 10 days or less
+      (soon)
+    ) {
+      return true;
+    } else {
+      return false;
+    }
+  },
+
+  @alias('isTrial') showTrialBanner: null,
+
+  @computed('isPaid', 'checkLicenseBanner')
+  showLicenseBanner(isPaid, check) {
+    return (isPaid && check);
+  },
+
+  @computed('isPaid', 'almostExceedingSeats', 'exceedingSeats')
+  showSeatsBanner(isPaid, almostExceeding, exceeding) {
+    return (isPaid && (almostExceeding || exceeding));
+  },
+
+  @computed('expiresSoon')
+  licenseClass(expiresSoon) {
+    if (expiresSoon) return 'alert';
+  },
+
+  trialClass: null,
+  seatsClass: 'alert',
+  paidClass: 'alert',
+
+  actions: {
+    closeLicenseBanner() {
+      this.get('storage').setItem(this.get('key'), this.get('daysUntilExpiry'));
+      return this.$('.enterprise-banner-license').remove();
```

Comment:
It’s preferable to let Ember handle managing the DOM rather than using jQuery to manipulate it, so you could accomplish this in two ways:

1. `this.set('showLicenseBanner', false);`, but that might feel a bit weird, overriding a computed property like that.

2. Adding a `licenseBannerDismissed` (or whatever) property, adding it to the `showLicenseBanner` as `return isPaid && check && !licenseBannerDismissed`, and changing this to `this.set('licenseBannerDismissed', true)`.

Reply:
Ah nice, I knew we'd end up doing things unidiomatically, good to know. Will amend this.